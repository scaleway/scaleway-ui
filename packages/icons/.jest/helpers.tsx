import { ThemeProvider } from '@emotion/react'
import makeHelpers from '@scaleway/jest-helpers'
import type { ReactNode } from 'react'
import { theme } from '@ultraviolet/ui'

type WrapperProps = {
  theme?: typeof theme
  children: ReactNode
}

const Wrapper = ({ theme, children }: WrapperProps) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

export const {
  renderWithTheme,
  shouldMatchEmotionSnapshot,
  shouldMatchEmotionSnapshotWithPortal,
} = makeHelpers(Wrapper, { classNameReplacer: className => className })
