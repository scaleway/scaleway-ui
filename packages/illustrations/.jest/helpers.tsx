import { ThemeProvider } from '@emotion/react'
import makeHelpers from '@scaleway/jest-helpers'
import type { ReactNode } from 'react'
import { consoleLightTheme, consoleDarkTheme } from '@ultraviolet/themes'

type WrapperProps = {
  theme?: typeof consoleLightTheme
  children: ReactNode
}
type DarkWrapperProps = {
  theme?: typeof consoleDarkTheme
  children: ReactNode
}

export const Wrapper = ({
  theme = consoleLightTheme,
  children,
}: WrapperProps) => <ThemeProvider theme={theme}>{children}</ThemeProvider>

export const DarkWrapper = ({
  theme = consoleDarkTheme,
  children,
}: DarkWrapperProps) => <ThemeProvider theme={theme}>{children}</ThemeProvider>

const lightThemeFunctions = makeHelpers(Wrapper, {
  classNameReplacer: className => className,
})
const darkThemeFunctions = makeHelpers(DarkWrapper, {
  classNameReplacer: className => className,
})

export const shouldMatchEmotionSnapshotLight =
  lightThemeFunctions.shouldMatchEmotionSnapshot
export const shouldMatchEmotionSnapshotDark =
  darkThemeFunctions.shouldMatchEmotionSnapshot
