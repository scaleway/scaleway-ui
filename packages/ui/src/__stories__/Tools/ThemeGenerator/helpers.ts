// Function imported from https://github.com/PimpTrizkit/PJs/wiki/12.-Shade,-Blend-and-Convert-a-Web-Color-(pSBC.js)#stackoverflow-archive-begin to shade hexa colors
type ShadeHexColorType = (color: string, percent: number) => string

export const shadeHexColor: ShadeHexColorType = (color, percent) => {
  const f = parseInt(color.slice(1), 16)
  const t = percent < 0 ? 0 : 255
  const p = percent < 0 ? percent * -1 : percent
  // eslint-disable-next-line no-bitwise
  const R = f >> 16
  // eslint-disable-next-line no-bitwise
  const G = (f >> 8) & 0x00ff
  // eslint-disable-next-line no-bitwise
  const B = f & 0x0000ff

  return `#${(
    0x1000000 +
    (Math.round((t - R) * p) + R) * 0x10000 +
    (Math.round((t - G) * p) + G) * 0x100 +
    (Math.round((t - B) * p) + B)
  )
    .toString(16)
    .slice(1)}`
}

export const generateShadeContrast = (
  shadeKey: string,
  value: string,
  index: number,
) =>
  Number(shadeKey) < 900
    ? shadeHexColor(value, (1 / 15) * (index + 1))
    : shadeHexColor(value, -(1 / 15) * (index + 1))
