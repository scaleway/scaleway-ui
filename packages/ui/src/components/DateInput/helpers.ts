import { CURRENT_MONTH, CURRENT_YEAR } from './constants'

// First day of the month for a given year
export const getMonthFirstDay = (
  month = CURRENT_MONTH,
  year = CURRENT_YEAR,
) => {
  const firstDay = new Date(year, month - 1, 1).getDay()

  // Change so that a week starts on monday
  return firstDay === 0 ? 6 : firstDay - 1
}

export const addZero = (value: number) => `${value}`.padStart(2, '0')

export const getPreviousMonth = (month: number, year: number) => {
  if (month === 1) {
    return [12, year - 1]
  }

  return [month - 1, year]
}

export const getNextMonth = (month: number, year: number) => {
  if (month === 12) {
    return [1, year + 1]
  }

  return [month + 1, year]
}

// Checks if two date values are of the same month and year
export const isSameMonth = (date: Date, basedate = new Date()) =>
  basedate.getMonth() === date.getMonth() &&
  basedate.getFullYear() === date.getFullYear()

// (bool) Checks if two date values are the same day
export const isSameDay = (date: Date, basedate = new Date()) =>
  basedate.getDate() === date.getDate() &&
  basedate.getMonth() + 1 === date.getMonth() + 1 &&
  basedate.getFullYear() === date.getFullYear()

// Default format if none is provided
const getDateISO = (showMonthYearPicker: boolean, date?: Date) => {
  if (date) {
    if (showMonthYearPicker) {
      return [addZero(date.getMonth() + 1), date.getFullYear()].join('/')
    }

    return [
      addZero(date.getMonth() + 1),
      addZero(date.getDate()),
      date.getFullYear(),
    ].join('/')
  }

  return ''
}

export const formatValue = (
  computedValue: Date | null,
  computedRange: {
    start: Date | null
    end: Date | null
  } | null,
  showMonthYearPicker: boolean,
  selectsRange: boolean,
  format?: (value?: Date) => string | undefined,
) => {
  if (selectsRange && computedRange) {
    return format
      ? `${format(computedRange.start ?? undefined) ? `${format(computedRange.start ?? undefined)} - ` : ''}${format(computedRange.end ?? undefined) ?? ''}`
      : `${getDateISO(showMonthYearPicker, computedRange.start ?? undefined)}${computedRange.start ? ' - ' : ''}${getDateISO(showMonthYearPicker, computedRange.end ?? undefined)}`
  }

  if (computedValue && format) {
    return format(computedValue)
  }
  if (computedValue) {
    return getDateISO(showMonthYearPicker, computedValue)
  }

  return undefined
}
