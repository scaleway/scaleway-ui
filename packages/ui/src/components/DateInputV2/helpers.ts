export const POPUP_WIDTH = '264px'

// The current year
export const CURRENT_YEAR = new Date().getFullYear()

// The current month starting from 1 - 12
// 1 => January, 12 => December
export const CURRENT_MONTH = new Date().getMonth() + 1

// The current day 1-28/29/30/31

// Weeks displayed on calendar
export const CALENDAR_WEEKS = 6

// Number days in a month for a given year
export const getMonthDays = (month = CURRENT_MONTH, year = CURRENT_YEAR) => {
  const months30 = [4, 6, 9, 11] // 30 days in april, june, september and november
  const leapYear = year % 4 === 0

  if (month === 2) {
    return leapYear ? 29 : 28 // Days in february
  }

  return months30.includes(month) ? 30 : 31
}

// First day of the month for a given year from 1 (monday) - 7 (saturday)
export const getMonthFirstDay = (month = CURRENT_MONTH, year = CURRENT_YEAR) =>
  new Date(`${year}-${month}-01`).getDay() + 1

// Adds a zero for 1-digit months/days
export const addZero = (value: number) => `${value}`.padStart(2, '0')

export const getPreviousMonth = (month: number, year: number) => {
  if (month === 1) {
    return [12, year - 1]
  }

  return [month - 1, year]
}

export const getNextMonth = (month: number, year: number) => {
  if (month === 12) {
    return [1, year + 1]
  }

  return [month + 1, year]
}

// Checks if two date values are of the same month and year
export const isSameMonth = (date: Date, basedate = new Date()) =>
  basedate.getMonth() + 1 === date.getMonth() + 1 &&
  basedate.getFullYear() === date.getFullYear()

// (bool) Checks if two date values are the same day
export const isSameDay = (date: Date, basedate = new Date()) =>
  basedate.getDate() === date.getDate() &&
  basedate.getMonth() + 1 === date.getMonth() + 1 &&
  basedate.getFullYear() === date.getFullYear()
