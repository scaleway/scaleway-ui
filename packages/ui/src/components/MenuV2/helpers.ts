import { Children, type ReactNode, isValidElement } from 'react'

/**
 * Search inside a children (React Element) recursively until a result is found
 */
export const searchChildren = (
  children: ReactNode,
  searchString: string,
): ReactNode[] => {
  const matches: ReactNode[] = []
  const searchRegex = new RegExp(searchString, 'i')

  Children.forEach(children, child => {
    if (typeof child === 'string' && child.match(searchRegex)) {
      matches.push(child)
    } else if (isValidElement(child)) {
      const childProps = child.props as { children: ReactNode }

      // Recursively search the children of this element
      const childMatches = searchChildren(childProps.children, searchString)

      if (childMatches.length > 0) {
        // If any matches are found within this child's children, push the entire child element
        matches.push(child)
      }
    }
  })

  return matches
}
