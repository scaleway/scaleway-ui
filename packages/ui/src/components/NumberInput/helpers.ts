export const bounded = (value: number, min: number, max: number) =>
  Math.max(min, Math.min(value, max))

export const roundStep = (value: number, step: number, direction: number) =>
  direction === -1
    ? Math.floor(value / step) * step
    : Math.ceil(value / step) * step

export const getMinusRoundedValue = (currentValue: number, step: number) =>
  currentValue % step === 0
    ? roundStep(currentValue - step, step, -1)
    : roundStep(currentValue, step, -1)

export const getPlusRoundedValue = (currentValue: number, step: number) =>
  currentValue % step === 0
    ? roundStep(currentValue + step, step, 1)
    : roundStep(currentValue, step, 1)
