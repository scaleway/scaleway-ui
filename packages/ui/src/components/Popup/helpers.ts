import type { RefObject } from 'react'

export type PopupPlacement = 'top' | 'right' | 'bottom' | 'left' | 'auto'
export const ARROW_WIDTH = 8 // in px
const SPACE = 4 // in px
const TOTAL_USED_SPACE = 0 // in px
export const DEFAULT_POSITIONS = {
  arrowLeft: -999,
  arrowTop: -999,
  arrowTransform: 'translate(-50%, -50)',
  placement: 'top',
  rotate: 135,
  popupInitialPosition: 'translate3d(-999px, -999px, 0)',
  popupPosition: 'translate3d(-999px, -999px, 0)',
}

type ComputePlacementTypes = {
  childrenStructuredRef: DOMRect
  popupStructuredRef: DOMRect
  popupPortalTarget: HTMLElement
  offsetParentRect: DOMRect
  offsetParent: Element
}

/**
 * This function will find the best placement in a window for popup based on children position and popup size
 */
const computePlacement = ({
  childrenStructuredRef,
  popupStructuredRef,
  offsetParentRect,
  offsetParent,
  popupPortalTarget,
}: ComputePlacementTypes) => {
  const {
    top: childrenTop,
    left: childrenLeft,
    right: childrenRight,
  } = childrenStructuredRef

  const {
    top: parentTop,
    left: parentLeft,
    right: parentRight,
  } = offsetParentRect

  const isPopupPortalTargetBody =
    popupPortalTarget === document.body || offsetParent === document.body

  const overloadedChildrenLeft = isPopupPortalTargetBody
    ? childrenLeft
    : childrenLeft - parentLeft
  const overloadedChildrenTop = isPopupPortalTargetBody
    ? childrenTop
    : childrenTop - parentTop
  const overloadedChildrenRight = isPopupPortalTargetBody
    ? childrenRight
    : childrenRight - parentRight

  const { width: popupWidth, height: popupHeight } = popupStructuredRef

  if (overloadedChildrenTop - popupHeight - TOTAL_USED_SPACE < 0) {
    return 'bottom'
  }

  if (overloadedChildrenLeft - popupWidth - TOTAL_USED_SPACE < 0) {
    return 'right'
  }

  if (
    overloadedChildrenRight + popupWidth + TOTAL_USED_SPACE >
    window.innerWidth
  ) {
    return 'left'
  }

  return 'top'
}

/**
 * This function will check if the offset parent is usable for popup positioning
 * If not it will loop and search for a compatible parent until document.body is reached
 */
const findOffsetParent = (element: RefObject<HTMLDivElement>) => {
  const offsetParent = element?.current?.offsetParent

  // We need to check if offsetParent is a table cell or a table because they are not suitable for positioning
  // https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetParent
  if (offsetParent && !['TH', 'TD', 'TABLE'].includes(offsetParent.tagName)) {
    return offsetParent
  }

  let currentElement = element?.current

  while (currentElement && currentElement.tagName !== 'BODY') {
    const { position } = window.getComputedStyle(currentElement)

    // Check if the current element is a potential offset parent
    if (position !== 'static') {
      return currentElement
    }

    currentElement = currentElement.parentElement as HTMLDivElement
  }

  // If no suitable offset parent is found, return the body element
  return document.body
}

type ComputePositionsTypes = {
  placement: PopupPlacement
  childrenRef: RefObject<HTMLDivElement>
  popupRef: RefObject<HTMLDivElement>
  popupPortalTarget: HTMLElement
}

/**
 * This function will compute the positions of popup and arrow based on children position and popup size
 */
export const computePositions = ({
  placement,
  childrenRef,
  popupRef,
  popupPortalTarget,
}: ComputePositionsTypes) => {
  const childrenRect = (
    childrenRef.current as HTMLDivElement
  ).getBoundingClientRect()
  const offsetParent = findOffsetParent(childrenRef)
  const offsetParentRect = offsetParent?.getBoundingClientRect() ?? {
    top: 0,
    left: 0,
    right: 0,
  }
  const popupStructuredRef = (
    popupRef.current as HTMLDivElement
  ).getBoundingClientRect()

  const placementBasedOnWindowSize =
    placement === 'auto'
      ? computePlacement({
          childrenStructuredRef: childrenRect,
          popupStructuredRef,
          offsetParentRect,
          popupPortalTarget,
          offsetParent,
        })
      : placement

  const {
    top: childrenTop,
    left: childrenLeft,
    right: childrenRight,
    width: childrenWidth,
    height: childrenHeight,
  } = childrenRect

  const {
    top: parentTop,
    left: parentLeft,
    right: parentRight,
  } = offsetParentRect

  const { width: popupWidth, height: popupHeight } = popupStructuredRef

  // offSetParent is the closest positioned ancestor. If the element is not positioned, the nearest table cell or root element is used.
  const isPopupPortalTargetBody =
    popupPortalTarget === document.body || offsetParent === document.body

  // It will get how much scroll is done on the page to compute the position of the popup
  const scrollTopValue = isPopupPortalTargetBody
    ? document.documentElement.scrollTop
    : offsetParent.scrollTop

  // We need to compute the position of the popup based on the parent element in the case the popup is not in the body
  const overloadedChildrenLeft = isPopupPortalTargetBody
    ? childrenLeft
    : childrenLeft - parentLeft
  const overloadedChildrenTop = isPopupPortalTargetBody
    ? childrenTop
    : childrenTop - parentTop
  const overloadedChildrenRight = isPopupPortalTargetBody
    ? childrenRight
    : childrenRight + childrenWidth + ARROW_WIDTH + SPACE - parentRight / 2

  switch (placementBasedOnWindowSize) {
    case 'bottom': {
      const positionX =
        overloadedChildrenLeft + childrenWidth / 2 - popupWidth / 2
      const positionY =
        overloadedChildrenTop +
        scrollTopValue +
        childrenHeight +
        ARROW_WIDTH +
        SPACE

      return {
        arrowLeft: popupWidth / 2,
        arrowTop: -ARROW_WIDTH - 5,
        arrowTransform: '',
        placement: 'bottom',
        rotate: 180,
        popupInitialPosition: `translate3d(${positionX}px, ${
          positionY - TOTAL_USED_SPACE
        }px, 0)`,
        popupPosition: `translate3d(${positionX}px, ${positionY}px, 0)`,
      }
    }
    case 'left': {
      const positionX =
        overloadedChildrenLeft - popupWidth - ARROW_WIDTH - SPACE * 2
      const positionY =
        overloadedChildrenTop +
        scrollTopValue -
        popupHeight / 2 +
        childrenHeight / 2

      return {
        arrowLeft: popupWidth + ARROW_WIDTH + 5,
        arrowTop: popupHeight / 2,
        arrowTransform: 'translate(-50%, -50%)',
        placement: 'left',
        rotate: -90,
        popupInitialPosition: `translate3d(${
          positionX + TOTAL_USED_SPACE
        }px, ${positionY}px, 0)`,
        popupPosition: `translate3d(${positionX}px, ${positionY}px, 0)`,
      }
    }
    case 'right': {
      const positionX = overloadedChildrenRight + ARROW_WIDTH + SPACE * 2
      const positionY =
        overloadedChildrenTop +
        scrollTopValue -
        popupHeight / 2 +
        childrenHeight / 2

      return {
        arrowLeft: -ARROW_WIDTH - 5,
        arrowTop: popupHeight / 2,
        arrowTransform: 'translate(50%, -50%)',
        placement: 'right',
        rotate: 90,
        popupInitialPosition: `translate3d(${
          positionX - TOTAL_USED_SPACE
        }px, ${positionY}px, 0)`,
        popupPosition: `translate3d(${positionX}px, ${positionY}px, 0)`,
      }
    }
    default: {
      // top placement is default value
      const positionX =
        overloadedChildrenLeft + childrenWidth / 2 - popupWidth / 2
      const positionY =
        overloadedChildrenTop +
        scrollTopValue -
        popupHeight -
        ARROW_WIDTH -
        SPACE

      return {
        arrowLeft: popupWidth / 2,
        arrowTop: popupHeight - 1,
        arrowTransform: '',
        placement: 'top',
        rotate: 0,
        popupInitialPosition: `translate3d(${positionX}px, ${
          positionY + TOTAL_USED_SPACE
        }px, 0)`,
        popupPosition: `translate3d(${positionX}px, ${positionY}px, 0)`,
      }
    }
  }
}
