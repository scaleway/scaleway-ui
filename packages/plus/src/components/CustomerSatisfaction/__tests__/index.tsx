import { describe, it, jest } from '@jest/globals'
import { screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { CustomerSatisfaction } from '..'
import { shouldMatchEmotionSnapshot } from '../../../../.jest/helpers'

describe('CustomerSatisfaction', () => {
  it('should work with parameters', () =>
    shouldMatchEmotionSnapshot(
      <CustomerSatisfaction value={4} onChange={jest.fn<any>()} />,
    ))

  it('should check hover and unhover', async () => {
    await shouldMatchEmotionSnapshot(
      <CustomerSatisfaction
        value={4}
        onChange={() => {}}
        data-testid="customer-satisfaction"
      />,
      {
        transform: async () => {
          await userEvent.hover(screen.getByTestId('customer-satisfaction-1'))
          await userEvent.unhover(screen.getByTestId('customer-satisfaction-1'))
        },
      },
    )
  })
})
