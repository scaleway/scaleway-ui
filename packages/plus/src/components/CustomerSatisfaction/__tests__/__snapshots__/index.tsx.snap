// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerSatisfaction should check hover and unhover 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from, 10%, 40%, 80%, to {
    -webkit-transform: translate3d(0, 0, 0);
    -moz-transform: translate3d(0, 0, 0);
    -ms-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  30% {
    -webkit-transform: translate3d(0, 5px, 0);
    -moz-transform: translate3d(0, 5px, 0);
    -ms-transform: translate3d(0, 5px, 0);
    transform: translate3d(0, 5px, 0);
    -webkit-transform: rotate(5deg);
    -moz-transform: rotate(5deg);
    -ms-transform: rotate(5deg);
    transform: rotate(5deg);
  }

  60% {
    -webkit-transform: translate3d(0, -6px, 0);
    -moz-transform: translate3d(0, -6px, 0);
    -ms-transform: translate3d(0, -6px, 0);
    transform: translate3d(0, -6px, 0);
  }

  90% {
    -webkit-transform: translate3d(0, -2px, 0);
    -moz-transform: translate3d(0, -2px, 0);
    -ms-transform: translate3d(0, -2px, 0);
    transform: translate3d(0, -2px, 0);
  }
}

.css-1cjs2t5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.css-pal2x8-Image-scaled {
  height: 88px;
  width: 88px;
  cursor: pointer;
  -webkit-transition: -webkit-transform 1s cubic-bezier(0.19, 1, 0.22, 1);
  transition: transform 1s cubic-bezier(0.19, 1, 0.22, 1);
}

.css-pal2x8-Image-scaled:hover,
.css-pal2x8-Image-scaled:focus {
  -webkit-transform: scale(1.2);
  -moz-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

.css-137fro5-Image-scaled-Image {
  height: 88px;
  width: 88px;
  cursor: pointer;
  -webkit-transition: -webkit-transform 1s cubic-bezier(0.19, 1, 0.22, 1);
  transition: transform 1s cubic-bezier(0.19, 1, 0.22, 1);
  -webkit-animation: animation-0 1s ease infinite;
  animation: animation-0 1s ease infinite;
}

.css-137fro5-Image-scaled-Image:hover,
.css-137fro5-Image-scaled-Image:focus {
  -webkit-transform: scale(1.2);
  -moz-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

<div
    class="css-1cjs2t5 ehpbis70"
    data-testid="customer-satisfaction"
  >
    <img
      class="css-pal2x8-Image-scaled epmg9lb0"
      data-testid="customer-satisfaction-1"
      src="test-file-stub"
    />
    <img
      class="css-pal2x8-Image-scaled epmg9lb0"
      data-testid="customer-satisfaction-2"
      src="test-file-stub"
    />
    <img
      class="css-pal2x8-Image-scaled epmg9lb0"
      data-testid="customer-satisfaction-3"
      src="test-file-stub"
    />
    <img
      class="css-137fro5-Image-scaled-Image epmg9lb0"
      data-testid="customer-satisfaction-4"
      src="test-file-stub"
    />
    <img
      class="css-pal2x8-Image-scaled epmg9lb0"
      data-testid="customer-satisfaction-5"
      src="test-file-stub"
    />
  </div>
</DocumentFragment>
`;

exports[`CustomerSatisfaction should work with parameters 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  from, 10%, 40%, 80%, to {
    -webkit-transform: translate3d(0, 0, 0);
    -moz-transform: translate3d(0, 0, 0);
    -ms-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  30% {
    -webkit-transform: translate3d(0, 5px, 0);
    -moz-transform: translate3d(0, 5px, 0);
    -ms-transform: translate3d(0, 5px, 0);
    transform: translate3d(0, 5px, 0);
    -webkit-transform: rotate(5deg);
    -moz-transform: rotate(5deg);
    -ms-transform: rotate(5deg);
    transform: rotate(5deg);
  }

  60% {
    -webkit-transform: translate3d(0, -6px, 0);
    -moz-transform: translate3d(0, -6px, 0);
    -ms-transform: translate3d(0, -6px, 0);
    transform: translate3d(0, -6px, 0);
  }

  90% {
    -webkit-transform: translate3d(0, -2px, 0);
    -moz-transform: translate3d(0, -2px, 0);
    -ms-transform: translate3d(0, -2px, 0);
    transform: translate3d(0, -2px, 0);
  }
}

.css-1cjs2t5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.css-pal2x8-Image-scaled {
  height: 88px;
  width: 88px;
  cursor: pointer;
  -webkit-transition: -webkit-transform 1s cubic-bezier(0.19, 1, 0.22, 1);
  transition: transform 1s cubic-bezier(0.19, 1, 0.22, 1);
}

.css-pal2x8-Image-scaled:hover,
.css-pal2x8-Image-scaled:focus {
  -webkit-transform: scale(1.2);
  -moz-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

.css-137fro5-Image-scaled-Image {
  height: 88px;
  width: 88px;
  cursor: pointer;
  -webkit-transition: -webkit-transform 1s cubic-bezier(0.19, 1, 0.22, 1);
  transition: transform 1s cubic-bezier(0.19, 1, 0.22, 1);
  -webkit-animation: animation-0 1s ease infinite;
  animation: animation-0 1s ease infinite;
}

.css-137fro5-Image-scaled-Image:hover,
.css-137fro5-Image-scaled-Image:focus {
  -webkit-transform: scale(1.2);
  -moz-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

<div
    class="css-1cjs2t5 ehpbis70"
  >
    <img
      class="css-pal2x8-Image-scaled epmg9lb0"
      data-testid="undefined-1"
      src="test-file-stub"
    />
    <img
      class="css-pal2x8-Image-scaled epmg9lb0"
      data-testid="undefined-2"
      src="test-file-stub"
    />
    <img
      class="css-pal2x8-Image-scaled epmg9lb0"
      data-testid="undefined-3"
      src="test-file-stub"
    />
    <img
      class="css-137fro5-Image-scaled-Image epmg9lb0"
      data-testid="undefined-4"
      src="test-file-stub"
    />
    <img
      class="css-pal2x8-Image-scaled epmg9lb0"
      data-testid="undefined-5"
      src="test-file-stub"
    />
  </div>
</DocumentFragment>
`;
